app/src/main/java/com/theuhooi/uhooipicbook/modules/monsterlist/MonsterListRecyclerViewAdapter.kt

```diff
@@ -3,18 +3,17 @@ package com.theuhooi.uhooipicbook.modules.monsterlist
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.ImageView
-import android.widget.TextView
+import androidx.lifecycle.LifecycleOwner
 import androidx.recyclerview.widget.RecyclerView
-import coil.api.load
-import com.theuhooi.uhooipicbook.R
+import com.theuhooi.uhooipicbook.databinding.ItemMonsterListBinding
 import com.theuhooi.uhooipicbook.modules.monsterlist.MonsterListFragment.OnListFragmentInteractionListener
 import com.theuhooi.uhooipicbook.modules.monsterlist.entities.MonsterItem
-import kotlinx.android.synthetic.main.item_monster_list.view.*
+import com.theuhooi.uhooipicbook.modules.monsterlist.viewmodel.MonsterListViewModel
 
 class MonsterListRecyclerViewAdapter(
-    private val monsters: List<MonsterItem>,
-    private val listener: OnListFragmentInteractionListener?
+    private val listener: OnListFragmentInteractionListener?,
+    private val viewModel: MonsterListViewModel,
```

Comment:
ViewAdapterにViewModelを渡すのって一般的なのでしょうか？
↓のようにモンスターのLiveDataを渡すだけでも実現できるので、こっちのほうが無駄なものを渡さなくていい感じがします。
（まだViewModelやLiveDataについて理解しきれていないので、処理的にダメでしたらすみません）

```suggestion
    private val monsters: LiveData<List<MonsterItem>>,
```

Reply:
あ、その方がいいかもしれないです！
viewModelを渡さないのであればLiveDataだけでいいかと思います

Reply:
よかったです！一般的だったらこれでもいいと思いましたが、できるだけ不要なものは渡したくないですもんね…